### PR TITLE
[Model/Layer] New forward to support CB (CommonDecoder->DecoderBlock->DecoderLayer->Attention/MLP)

### PR DESCRIPTION
**Pull request by pujiang2018**

### CODE REVIEW

**Comment by pujiang2018** [comment]
@abenmao @Duyi-Wang pls help to review.

File: src/layers/decoder_block.h [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+        // Data preparation
+        std::vector<int> seqIDs(seqs.size());
+        size_t totInSeqLen = 0;
+        for (int i = 0; i < seqs.size(); ++i) {
+            seqIDs[i] = seqs[i]->getSequenceID();
+            totInSeqLen += seqs[i]->getInputSeqLen();
+        }
+
+        kvCacheMgr.prepareCache(seqIDs);
```
Comment (Duyi-Wang) [inline src/layers/decoder_block.h:79]: I prefer to prepare KVCache at a high level api.
We can assume all seqs' KVCache is ready.

File: src/models/common_decoder.h [hunk 2]
```diff
@@ -265,10 +254,6 @@ class CommonDecoder : public AbstractDecoder {
         if (this->attnMask) free(this->attnMask);
 
         delete this->predictor;
-
```
[the remark this answers is not in the document]
Comment (pujiang2018) [reply src/models/common_decoder.h:268]: Done

File: src/layers/decoder_block.h (hunk 1 above)
Comment (pujiang2018) [reply src/layers/decoder_block.h]: commented and added TODO

**Pull request merged**